### PR TITLE
use felix messages for ipam watcher

### PR DESCRIPTION
**Pull request by hedibouattour**

### CODE REVIEW

**Comment by hedibouattour** [comment]
depends on this PR in calico repo:
(just a wip PR in fork for now) https://github.com/calico-vpp/calico/pull/8

File: calico-vpp-agent/policy/policy_server.go [hunk 1]
```diff
@@ -1098,11 +1098,19 @@ func (s *Server) handleHostMetadataRemove(msg *proto.HostMetadataRemove, pending
 }
 
 func (s *Server) handleIpamPoolUpdate(msg *proto.IPAMPoolUpdate, pending bool) (err error) {
+	common.SendEvent(common.CalicoVppEvent{
+		Type: common.IpamPoolUpdate,
+		New:  msg,
+	})
 	s.log.Debugf("Ignoring IpamPoolUpdate")
```
Comment (sknat) [inline calico-vpp-agent/policy/policy_server.go:1105]: We are not `"Ignoring IpamPoolUpdate"` anymore 😄

File: calico-vpp-agent/policy/policy_server.go [hunk 2]
```diff
@@ -1098,11 +1098,19 @@ func (s *Server) handleHostMetadataRemove(msg *proto.HostMetadataRemove, pending
...
 	return nil
 }
 
 func (s *Server) handleIpamPoolRemove(msg *proto.IPAMPoolRemove, pending bool) (err error) {
+	common.SendEvent(common.CalicoVppEvent{
+		Type: common.IpamPoolRemove,
+		Old:  msg,
+	})
 	s.log.Debugf("Ignoring IpamPoolRemove")
```
Comment (sknat) [inline calico-vpp-agent/policy/policy_server.go:1114]: same here 😄

File: calico-vpp-agent/connectivity/connectivity_server.go [hunk 3]
```diff
@@ -269,7 +271,7 @@ func (s *ConnectivityServer) getProviderType(cn *common.NodeConnectivity) (strin
 	if ipPool == nil {
 		return FLAT, nil
 	}
-	if ipPool.Spec.IPIPMode == calicov3.IPIPModeAlways {
+	if strings.ToLower(ipPool.Pool.IpipMode) == strings.ToLower(string(calicov3.IPIPModeAlways)) {
```
Comment (sknat) [inline calico-vpp-agent/connectivity/connectivity_server.go:274]:
````go
import "github.com/projectcalico/calico/libcalico-go/lib/backend/encap"
(ipPool.Pool.IpipMode == string(encap.Always)
````

File: calico-vpp-agent/watchers/ipam.go [hunk 4]
```diff
@@ -83,7 +84,7 @@ func (c *ipamCache) ForceReady() {
 
 // match checks whether we have an IP pool which contains the given prefix.
 // If we have, it returns the pool.
-func (c *ipamCache) GetPrefixIPPool(prefix *net.IPNet) *calicov3.IPPool {
+func (c *ipamCache) GetPrefixIPPool(prefix *net.IPNet) *proto.IPAMPoolUpdate {
```
Comment (sknat) [inline calico-vpp-agent/watchers/ipam.go:87]: We probably could just return the `IPAMPool`

**Review by sknat — APPROVED** [review]

**Pull request merged**